### ISSUE TITLE
Crash with completion

### ISSUE THREAD

**Issue by sean-mcmanus** [microsoft/vscode-cpptools#4359]
I got the same issue on Mac OS (10.14.6)

Here is the crash log.
[Microsoft.VSCode.CPP.Extension.darwin_2019-09-24-105103_Jins-MacBook-Pro.crash.log](https://github.com/microsoft/vscode-cpptools/files/3647862/Microsoft.VSCode.CPP.Extension.darwin_2019-09-24-105103_Jins-MacBook-Pro.crash.log)

**This issue started from version `0.22.0`, I have tested all versions since `0.22.0`, all of them have this issue, and `0.21.0` works charmingly. Hope this information can help you guys locate the bug.**

@bobbrow @sean-mcmanus

_Originally posted by @jin-qin in https://github.com/microsoft/vscode-cpptools/issues/3235#issuecomment-534600613_

**Comment by sean-mcmanus** [comment]
@jin-qin Do you have any more repro details? Other than doing a completion, what other operation occurred beforehand? Does it happen all the time or randomly?

**Comment by jin-qin** [comment]
> @jin-qin Do you have any more repro details? Other than doing a completion, what other operation occurred beforehand? Does it happen all the time or randomly?

@sean-mcmanus 
It happens all the time.
Here are the operation steps:
1. Open VSCode
2. Open a header or source file with some c++ codes.
3. Wait a minute.

**Comment by sean-mcmanus** [comment]
There must be something causing the creation of our browse database to fail. Can you change the browse.databaseFilename setting to some location like "${workspaceFolder}/.vscode/vc.db" and make sure it is written? That should fix the crash, but we should also add some checks to avoid crashing if the database never got created.

**Comment by jin-qin** [comment]
> There must be something causing the creation of our browse database to fail. Can you change the browse.databaseFilename setting to some location like "${workspaceFolder}/.vscode/vc.db" and make sure it is written? That should fix the crash, but we should also add some checks to avoid crashing if the database never got created.

But version `0.21.0` works fine, and it has got trouble since version `0.22.0`, you can check the difference between version `0.21.0` and `0.22.0`

**Comment by bobbrow** [comment]
There are a lot of changes between versions, so this is not a trivial thing to do.  Can you try the workaround that @sean-mcmanus suggested and let us know if it fixes the problem?  That will help us have a better idea of where to look.

**Comment by jin-qin** [comment]
> There are a lot of changes between versions, so this is not a trivial thing to do. Can you try the workaround that @sean-mcmanus suggested and let us know if it fixes the problem? That will help us have a better idea of where to look.

I tried yesterday, could not work.
@sean-mcmanus @bobbrow

**Comment by sean-mcmanus** [comment]
You tried what? Did the database file not get created? Do you see any errors when you enable logging: https://code.visualstudio.com/docs/cpp/enable-logging-cpp

**Comment by jin-qin** [comment]
> You tried what? Did the database file not get created? Do you see any errors when you enable logging: https://code.visualstudio.com/docs/cpp/enable-logging-cpp

The crash log: [Microsoft.VSCode.CPP.Extension.darwin_2019-10-03-224158_Jins-MacBook-Pro.crash.log](https://github.com/microsoft/vscode-cpptools/files/3688759/Microsoft.VSCode.CPP.Extension.darwin_2019-10-03-224158_Jins-MacBook-Pro.crash.log)

I tried to do what you told me 3 days ago, and I did not see any database file get created, and here is the error info:

```
[Error - 10:41:58 PM] Connection to server got closed. Server will not be restarted.
[Error - 10:41:58 PM] Request textDocument/documentSymbol failed.
Error: Connection got disposed.
	at Object.dispose (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:2898:25)
	at Object.dispose (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:3193:35)
	at LanguageClient.handleConnectionClosed (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:5275:42)
	at LanguageClient.handleConnectionClosed (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:15682:15)
	at closeHandler (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:5262:18)
	at CallbackList.invoke (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:7905:39)
	at Emitter.fire (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:7963:36)
	at closeHandler (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:2248:26)
	at CallbackList.invoke (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:7905:39)
	at Emitter.fire (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:7963:36)
	at StreamMessageReader.fireClose (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:12091:27)
	at Socket.module.exports.listen.readable.on (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:12131:46)
	at Socket.emit (events.js:187:15)
	at Pipe.Socket._destroy._handle.close (net.js:606:12)
[Error - 10:41:58 PM] Request textDocument/documentSymbol failed.
Error: Connection got disposed.
	at Object.dispose (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:2898:25)
	at Object.dispose (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:3193:35)
	at LanguageClient.handleConnectionClosed (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:5275:42)
	at LanguageClient.handleConnectionClosed (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:15682:15)
	at closeHandler (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:5262:18)
	at CallbackList.invoke (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:7905:39)
	at Emitter.fire (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:7963:36)
	at closeHandler (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:2248:26)
	at CallbackList.invoke (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:7905:39)
	at Emitter.fire (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:7963:36)
	at StreamMessageReader.fireClose (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:12091:27)
	at Socket.module.exports.listen.readable.on (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:12131:46)
	at Socket.emit (events.js:187:15)
	at Pipe.Socket._destroy._handle.close (net.js:606:12)
[Error - 10:41:58 PM] Request textDocument/documentSymbol failed.
Error: Connection got disposed.
	at Object.dispose (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:2898:25)
	at Object.dispose (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:3193:35)
	at LanguageClient.handleConnectionClosed (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:5275:42)
	at LanguageClient.handleConnectionClosed (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:15682:15)
	at closeHandler (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:5262:18)
	at CallbackList.invoke (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:7905:39)
	at Emitter.fire (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:7963:36)
	at closeHandler (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:2248:26)
	at CallbackList.invoke (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:7905:39)
	at Emitter.fire (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:7963:36)
	at StreamMessageReader.fireClose (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:12091:27)
	at Socket.module.exports.listen.readable.on (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:12131:46)
	at Socket.emit (events.js:187:15)
	at Pipe.Socket._destroy._handle.close (net.js:606:12)
[Error - 10:41:58 PM] Request textDocument/documentSymbol failed.
Error: Connection got disposed.
	at Object.dispose (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:2898:25)
	at Object.dispose (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:3193:35)
	at LanguageClient.handleConnectionClosed (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:5275:42)
	at LanguageClient.handleConnectionClosed (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:15682:15)
	at closeHandler (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:5262:18)
	at CallbackList.invoke (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:7905:39)
	at Emitter.fire (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:7963:36)
	at closeHandler (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:2248:26)
	at CallbackList.invoke (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:7905:39)
	at Emitter.fire (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:7963:36)
	at StreamMessageReader.fireClose (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:12091:27)
	at Socket.module.exports.listen.readable.on (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:12131:46)
	at Socket.emit (events.js:187:15)
	at Pipe.Socket._destroy._handle.close (net.js:606:12)
[Error - 10:41:58 PM] Request completionItem/resolve failed.
Error: Connection got disposed.
	at Object.dispose (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:2898:25)
	at Object.dispose (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:3193:35)
	at LanguageClient.handleConnectionClosed (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:5275:42)
	at LanguageClient.handleConnectionClosed (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:15682:15)
	at closeHandler (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:5262:18)
	at CallbackList.invoke (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:7905:39)
	at Emitter.fire (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:7963:36)
	at closeHandler (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:2248:26)
	at CallbackList.invoke (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:7905:39)
	at Emitter.fire (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:7963:36)
	at StreamMessageReader.fireClose (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:12091:27)
	at Socket.module.exports.listen.readable.on (/Users/jinqin/.vscode/extensions/ms-vscode.cpptools-0.25.1/dist/main.js:12131:46)
	at Socket.emit (events.js:187:15)
	at Pipe.Socket._destroy._handle.close (net.js:606:12)

```

**Comment by sean-mcmanus** [comment]
We need to figure out why our code isn't able to create our database. Can you enable debug logging (https://code.visualstudio.com/docs/cpp/enable-logging-cpp ) and see what errors it shows when you just open a C/C++ workspace? Under normal conditions you should see "Code browsing service intialized", "Discovering files...", and "tag parsing file:" messages (you may need to do a Reset IntelliSense Database command if your database is fully built).

**Comment by jin-qin** [comment]
> We need to figure out why our code isn't able to create our database. Can you enable debug logging (https://code.visualstudio.com/docs/cpp/enable-logging-cpp ) and see what errors it shows when you just open a C/C++ workspace? Under normal conditions you should see "Code browsing service intialized", "Discovering files...", and "tag parsing file:" messages (you may need to do a Reset IntelliSense Database command if your database is fully built).

FYI

```
initialized
workspace/didChangeConfiguration
IntelliSense Engine = Default.
Autocomplete is enabled.
Enhanced Colorization is enabled.
Error squiggles are enabled if all header dependencies are resolved.
File exclude: **/.git
File exclude: **/.svn
File exclude: **/.hg
File exclude: **/CVS
File exclude: **/.DS_Store
File exclude: **/.vscode
Search exclude: **/node_modules
Search exclude: **/bower_components
Search exclude: **/.vscode
Attempting to get defaults from compiler found on the machine: '/usr/bin/clang'
terminating child process: 1776
cpptools/queryCompilerDefaults: 1
Attempting to get defaults from compiler found on the machine: '/usr/bin/clang'
terminating child process: 1778
cpptools/activeDocumentChange
cpptools/textEditorSelectionChange
textDocument/didOpen
textDocument/codeAction: 3
idle loop: reparsing the active document
cpptools/textEditorSelectionChange
textDocument/codeAction: 4
```

**Comment by sean-mcmanus** [comment]
Yeah, something is causing the database creation to fail (turns out we don't log a user-visible message in that case). I got a repro via denying permissions to the database folder. Can you double-check that you have access to the database and can create files at the databaseFilename location?

We've made some fixes for our pending 0.26.0-insiders3 that should fix the crashing, but you'll see "Initializing..." instead if the database can't get created.

**Comment by jin-qin** [comment]
> Yeah, something is causing the database creation to fail (turns out we don't log a user-visible message in that case). I got a repro via denying permissions to the database folder. Can you double-check that you have access to the database and can create files at the databaseFilename location?
> 
> We've made some fixes for our pending 0.26.0-insiders3 that should fix the crashing, but you'll see "Initializing..." instead if the database can't get created.

Well, I believe I have the permission to create files, but I am not sure if I can access to the database, because I have never seen such a database file in the workspace.

**Comment by sean-mcmanus** [comment]
We've fixed the crashes with https://github.com/microsoft/vscode-cpptools/releases/tag/0.26.0-insiders3 . Are you still seeing crashes? You should probably see "Initializing..." instead...not sure if that solves your issue of failing to create the database though.
